providers/route53/route53Provider.go

```diff
@@ -12,28 +13,42 @@ import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/private/waiter"
 	r53 "github.com/aws/aws-sdk-go/service/route53"
+	r53d "github.com/aws/aws-sdk-go/service/route53domains"
 	"github.com/pkg/errors"
 )
 
 type route53Provider struct {
-	client *r53.Route53
-	zones  map[string]*r53.HostedZone
+	client    *r53.Route53
+	registrar *r53d.Route53Domains
+	zones     map[string]*r53.HostedZone
 }
 
-func newRoute53(m map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
+func newRoute53Reg(conf map[string]string) (providers.Registrar, error) {
+	return newRoute53(conf, nil)
+}
+
+func newRoute53Dsp(conf map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
+	return newRoute53(conf, metadata)
+}
+
+func newRoute53(m map[string]string, metadata json.RawMessage) (*route53Provider, error) {
 	keyId, secretKey := m["KeyId"], m["SecretKey"]
 
+	// Route53 uses a global endpoint and route53domains
+	// currently only has a single regional endpoint in us-east-1
+	// http://docs.aws.amazon.com/general/latest/gr/rande.html#r53_region
 	config := &aws.Config{
-		Region: aws.String("us-west-2"),
+		Region: aws.String("us-east-1"),
 	}
 
 	if keyId != "" || secretKey != "" {
 		config.Credentials = credentials.NewStaticCredentials(keyId, secretKey, "")
 	}
 	sess := session.New(config)
 
-	api := &route53Provider{client: r53.New(sess)}
+	api := &route53Provider{client: r53.New(sess), registrar: r53d.New(sess)}
```

Comment:
I think this may run into issues if you are only using domains or only using r53, depending on the access level of the token. Perhaps there should be a seperate type for registrar and dns providers?